### PR TITLE
Send token in Send page

### PR DESCRIPTION
**Pull request by oneleo**
# Send token in Send page

## Demo

([Video](https://youtu.be/ICW6kxak9nk))

### CODE REVIEW

File: app/page/info/tokens.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
-import { formatUnits, getAddress, toNumber } from "ethers"
+import { getAddress, toNumber } from "ethers"
 import { useCallback, useState, type ChangeEvent, type FormEvent } from "react"
 
 import { useProviderContext } from "~app/context/provider"
 import { useAccount, useAction, useTokens } from "~app/storage"
-import { getChainName, getErc20Contract } from "~packages/network/util"
+import {
+  formatUnitsToFixed,
```
Comment (pilagod) [inline app/page/info/tokens.tsx:9]: nit: This would be better to put in number util

File: app/page/send.tsx [hunk 2]
```diff
@@ -43,17 +75,51 @@ export function Send() {
...
+            <option key={nativeToken.address} value={nativeToken.address}>
+              {nativeToken.symbol}: {nativeToken.balance} {nativeToken.symbol}
+            </option>
+            {tokens.map((token) => {
+              const balance = parseFloat(
+                ethers.formatUnits(
+                  ethers.toBeHex(token.balance),
+                  ethers.toNumber(token.decimals)
+                )
+              ).toFixed(6)
```
Comment (pilagod) [inline app/page/send.tsx:114]: nit: Why not using `formatUnitsToFixed` util?

File: app/page/send.tsx [hunk 3]
```diff
@@ -5,17 +5,49 @@ import { Link } from "wouter"
...
   const account = useAccount()
-  const [txHash, setTxHash] = useState<HexString>("")
+  const tokens = useTokens()
+
+  const nativeToken: Token = {
+    address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+    symbol: `${getChainName(account.chainId)}ETH`,
+    decimals: 18,
+    balance: formatUnitsToFixed(account.balance, 18)
+  }
```
Comment (pilagod) [inline app/page/send.tsx:28]:
```suggestion
  const nativeToken: Token = {
    address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
    symbol: `${getChainName(account.chainId)}ETH`,
    decimals: 18,
    balance: formatUnitsToFixed(account.balance, 18)
  }
  const tokens = [
      nativeToken,
      ...useTokens()
  ]
```

File: app/page/send.tsx [hunk 4]
```diff
@@ -5,17 +5,49 @@ import { Link } from "wouter"
...
+    console.log(`tokenAddress: ${tokenAddress}`)
+    if (address.isEqual(nativeToken.address, tokenAddress)) {
+      setToken(nativeToken)
+      setIsEth(true)
+      return
+    }
+    const token = tokens.find((token) => token.address === tokenAddress)
+    setToken(token)
+    setIsEth(false)
+  }
```
Comment (pilagod) [inline app/page/send.tsx:50]:
```suggestion
  const handleAssetChange = async (event: ChangeEvent<HTMLSelectElement>) => {
    const tokenAddress = event.target.value
    console.log(`tokenAddress: ${tokenAddress}`)
    const token = tokens.find((token) => token.address === tokenAddress)
    setToken(token)
  }
```

File: app/page/send.tsx [hunk 5]
```diff
@@ -43,17 +75,51 @@ export function Send() {
 
   const handleSend = useCallback(async () => {
```
Comment (pilagod) [inline app/page/send.tsx:76]:
```suggestion
  const handleSend = useCallback(async () => {
      if (token.address === nativeToken.address) {
          return sendNativeToken(...)
      }
      return sendErc20Token(...)
  })
```

File: app/page/send.tsx (hunk 2 above)
Comment (oneleo) [reply app/page/send.tsx:114]: This was my oversight, let me fix it!

**Review by pilagod — APPROVED** [review]

**Pull request merged**